pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,80 @@
+package prometheus
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const prometheusDefaultPort = "2112"
+
+var (
+	ApiCallsFailure = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "api_call_failures_total",
+		Help: "The total number of api calls that failed by the updater",
+	}, []string{"node", "errorType"})
+
+	OperationDelay = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "operation_delay",
+			Help: "Represent the latency of the update operation",
+		}, []string{"node"})
+)
+
+func extractErrorType(error string) string {
+	// TODO temporary until PodresourceAPI errors will be standardize
+	if strings.Contains(error, "GetAllocatableResources disabled") {
```

Comment:
I'm always wary when we make logic on the error text. I'd just add a label to track the API being called, and record the raw number of errors. We will differentiate later, if needed.

Reply:
We are tracking the number of  API calls using the metric itself (each call increase the metric by 1).
The label is in order to differentiate between feature-gate/non-feature-gate errors.
AFAIU we don't need a label to track the general number of calls because that is to the whole purpose of the metric itself.

Reply:
then how we can report the `List` errors, which must always be zero (otherwise something horrible is happening)?

Reply:
Probably I'm missing something.
The ```prometheus.ApiCallsFailure``` should track all types of API failures (any type of call to the API server, ```List```, ```Get``` etc. ) or just the PodreasourceAPI calls?

Reply:
This metric must count the failures of all the `podresources` API calls, thus `List` and `GetAllocatableResources`. The former count (`List`) must always be zero, and we must report this.

Reply:
Summarize it here for documentation purposes:
There are two types of podresourceapi calls:
List https://github.com/kubernetes/kubernetes/blob/release-1.22/pkg/kubelet/apis/podresources/server_v1.go#L50 
and GetAllocatableResources https://github.com/kubernetes/kubernetes/blob/release-1.22/pkg/kubelet/apis/podresources/server_v1.go#L82